tck/features/SemanticErrorAcceptance.feature

```diff
@@ -0,0 +1,404 @@
+      RETURN r
+      """
+    Then a SyntaxError should be raised at compile time: VariableTypeConflict
+
+  Scenario: Failing when a relationship is used as a node
+    When executing query:
+      """
+      MATCH ()-[r]-(r)
+      RETURN r
+      """
+    Then a SyntaxError should be raised at compile time: VariableTypeConflict
+
+  Scenario: Failing when using `type()` on a node
+    When executing query:
+      """
+      MATCH (r)
+      RETURN type(r)
+      """
+    Then a SyntaxError should be raised at compile time: InvalidArgumentType
+
+  Scenario: Failing when a using `length()` on a node
+    When executing query:
+      """
+      MATCH (r)
+      RETURN length(r)
+      """
+    Then a SyntaxError should be raised at compile time: InvalidArgumentType
+
+  Scenario: Failing when re-using a relationship in the same pattern
+    When executing query:
+      """
+      MATCH (a)-[r]->()-[r]->(a)
+      RETURN r
+      """
+    Then a SyntaxError should be raised at compile time: RelationshipUniquenessViolation
+
+  Scenario: Failing when using NOT on string literal
+    When executing query:
+      """
+      RETURN NOT 'foo'
+      """
+    Then a SyntaxError should be raised at compile time: InvalidArgumentType
+
+  Scenario: Failing when using variable length relationship in CREATE
+    When executing query:
+      """
+      CREATE ()-[:FOO*2]->()
+      """
+    Then a SyntaxError should be raised at compile time: CreatingVarLength
+
+  Scenario: Failing when using variable length relationship in MERGE
+    When executing query:
+      """
+      MERGE ()-[:FOO*2]->()
+      """
+    Then a SyntaxError should be raised at compile time: CreatingVarLength
+
+  Scenario: Failing when using parameter as node predicate in MATCH
+    When executing query:
+      """
+      MATCH (n {param})
+      RETURN n
+      """
+    Then a SyntaxError should be raised at compile time: InvalidParameterUse
+
+  Scenario: Failing when using parameter as relationship predicate in MATCH
+    When executing query:
+      """
+      MATCH ()-[r:FOO {param}]->()
+      RETURN r
+      """
+    Then a SyntaxError should be raised at compile time: InvalidParameterUse
+
+  Scenario: Failing when using parameter as node predicate in MERGE
+    When executing query:
+      """
+      MERGE (n {param})
+      RETURN n
+      """
+    Then a SyntaxError should be raised at compile time: InvalidParameterUse
+
+  Scenario: Failing when using parameter as relationship predicate in MERGE
+    When executing query:
+      """
+      MERGE ()-[r:FOO {param}]->()
+      RETURN r
+      """
+    Then a SyntaxError should be raised at compile time: InvalidParameterUse
+
+  Scenario: Failing when deleting an integer expression
+    When executing query:
+      """
+      MATCH ()
+      DELETE 1 + 1
+      """
+    Then a SyntaxError should be raised at compile time: InvalidArgumentType
+
+  Scenario: Failing when using CREATE on already bound node
```

Comment:
"Failing when using CREATE on already bound node" -> "Failing when using CREATE on a node that is already bound"